tcms/static/js/testrun_actions.js

```diff
@@ -244,63 +252,42 @@ Nitrate.TestRuns.AssignCase.on_load = function() {
   });
 };
 
+function updateExecutionStatus(executionId, caseId, executionStatusPk) {
 
-function updateRunStatus(object_pk, value, callback) {
-  jQ.ajax({
-    'url': '/runs/case-run-update-status/',
-    'type': 'POST',
-    'data': {'object_pk': object_pk, 'status_id': value },
-    'success': function (data, textStatus, jqXHR) {
-      callback();
-    },
-    'error': function (jqXHR, textStatus, errorThrown) {
-      json_failure(jqXHR);
-    }
-  });
-}
+  const container = $('.update_form').parents().eq(3)
+  const parent = container.parent()
+  const title = parent.prev()
+  const link = title.find('.expandable')[0];
 
-var updateCaseRunStatus = function(e) {
-  e.stopPropagation();
-  e.preventDefault();
-  var container = jQ(this).parents().eq(3);
-  var parent = container.parent();
-  var title = parent.prev();
-  var link = title.find('.expandable')[0];
-  var parameters = Nitrate.Utils.formSerialize(this);
-  var object_pk = parameters['object_pk'];
-  var value = parameters['value'];
-
-  // Callback when
-  var callback = function(t) {
-    // Update the contents
-    const value = parameters['value']
-    if (value != '') {
-      // Update the case run status icon
-      var crs = Nitrate.TestRuns.CaseRunStatus;
-      title.find('.execution_status_icon').each(function(index) {
-        for (i in crs) {
-          jQ(this).removeClass(crs[i].icon);
-        }
-        const execution = Nitrate.TestRuns.CaseRunStatus.find(tcs => tcs.pk === value);
-        jQ(this).addClass(execution.icon);
-        jQ(this).css('color', execution.color)
-      });
+  addComment(executionId, caseId)
 
-      // Update related people
-      var usr = Nitrate.User;
-      title.find('.link_tested_by').each(function(i) {
-        jQ(this).html(usr.username);
-      });
-    }
+  // Update the object when changing the status
+  jsonRPC('TestExecution.update', [executionId, {
+    'status': executionStatusPk,
+    'tested_by': Nitrate.User.pk
+  }], data => {
+
+    // Update the case run status icon
+    const executionStatusIcon = $(title.find('.execution_status_icon'))
 
+    // remove all the icon classes from the icon, so that we can add the proper class few lines below
+    executionStatuses.forEach(status => executionStatusIcon.removeClass(status.icon))
+
+    // update the icon in the table row
+    const executionStatus = executionStatuses.find(status => status.id === parseInt(executionStatusPk))
+    executionStatusIcon.addClass(executionStatus.icon)
+    executionStatusIcon.css('color', executionStatus.color)
+
+    // Update related people
+    title.find('.link_tested_by').html(data.tested_by);
     // Mark the case run to mine
     if (!title.is('.mine')) {
       title.addClass('mine');
     }
 
     // Blind down next case
     fireEvent(link, 'click');
-    if (jQ('#id_check_box_auto_blinddown').attr('checked') && parameters['value'] != '') {
+    if ($('#id_check_box_auto_blinddown').attr('checked') && executionStatusPk) {
```

Comment:
something here still doesn't work for me. When I manage to update the statuses the next TE row is not expanded automatically if I have the checkbox at the top selected.

Reply:
turns out `jQ` and `$` return different things in this case